app/code/Magento/InventoryInStorePickup/Model/IsOrderReadyForPickup.php

```diff
@@ -28,29 +31,70 @@ class IsOrderReadyForPickup implements IsOrderReadyForPickupInterface
      */
     private $orderRepository;
 
+    /**
+     * @var OperationResultInterfaceFactory
+     */
+    private $operationResultFactory;
+
     /**
      * @param IsFulfillable $isFulfillable
      * @param OrderRepositoryInterface $orderRepository
+     * @param OperationResultInterfaceFactory $operationResultFactory
      */
     public function __construct(
         IsFulfillable $isFulfillable,
-        OrderRepositoryInterface $orderRepository
+        OrderRepositoryInterface $orderRepository,
+        OperationResultInterfaceFactory $operationResultFactory
     ) {
         $this->isFulfillable = $isFulfillable;
         $this->orderRepository = $orderRepository;
+        $this->operationResultFactory = $operationResultFactory;
     }
 
     /**
      * Check if order can be shipped and the pickup location has enough QTY.
      *
-     * @param int $orderId
-     * @return bool
+     * @param int[] $orderIds
+     * @return OperationResultInterface
      */
-    public function execute(int $orderId): bool
+    public function execute(array $orderIds): OperationResultInterface
     {
-        $order = $this->orderRepository->get($orderId);
+        $success = [];
+        $errors = [];
+        foreach ($orderIds as $orderId) {
+            try {
+                $order = $this->orderRepository->get($orderId);
+            } catch (NoSuchEntityException $exception) {
+                $errors[] = [
+                    'order_id' => $orderId,
+                    'message' => $exception->getMessage()
+                ];
+                continue;
+            }
+            if (!$this->canShip($order)) {
+                $errors[] = [
+                    'order_id' => $orderId,
+                    'message' => __('Cannot ship the order.')
+                ];
+                continue;
+            }
+
+            if (!$this->isFulfillable->execute($order)) {
+                $errors[] = [
+                    'order_id' => $orderId,
+                    'message' => __('Cannot fulfill the order.')
+                ];
+                continue;
+            }
+            $success[] = $orderId;
+        }
 
-        return $this->canShip($order) && $this->isFulfillable->execute($order);
+        return $this->operationResultFactory->create(
```

Comment:
Response used to be `true` and `false`. I would expect same behavior when multiple orders are requested.

This implementation never returns `false`.